lib/facter/swap_file_size.rb

```diff
@@ -0,0 +1,19 @@
+Facter.add('swap_file_size') do
```

Comment:
Yep, that's what I did here https://github.com/petems/petems-swap_file/blob/add_swapfilesizes_fact/lib/facter/swap_file_size.rb for #47 

For dynamic naming, it's Ruby so you can do something like this:

```
if Facter::Util::Resolution.which('rpm')
  package_data = Facter::Util::Resolution.exec('rpm --query --all  --qf "%{NAME}||%{VERSION}\n"')

  package_data.split("\n").each do |line|

    rpm_detail      = line.split("||")

    package_name    = 'package_' + rpm_detail[0]
    package_version = rpm_detail[1]

    Facter.add(package_name) do
      setcode do
        package_version
      end
    end

  end
end
```